moveit_ros/moveit_servo/test/test_parameter_struct.hpp

```diff
@@ -0,0 +1,130 @@
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/*      Title     : test_parameter_struct.hpp
+ *      Project   : moveit_servo
+ *      Created   : 07/23/2020
+ *      Author    : Adam Pettinger
+ *      Desc      : Returns a ServoParameters object for testing
+ */
+
+#pragma once
+
+#include <moveit_servo/servo_parameters.h>
+
+/** \brief Returns a shared ptr to a ServoParams object with valid (and 'default') fields
+   * A ServoParameters of this type should match what is loaded from config/panda_simulated_config.yaml
+   */
+moveit_servo::ServoParametersPtr getTestParameters()
+{
+  auto output = std::make_shared<moveit_servo::ServoParameters>();
+
+  // Populate the fields
+  output->use_gazebo = true;
+  output->status_topic = "servo_server/status";
+  output->cartesian_command_in_topic = "servo_server/delta_twist_cmds";
+  output->joint_command_in_topic = "servo_server/delta_joint_cmds";
+  output->robot_link_command_frame = "panda_link0";
+  output->command_in_type = "unitless";
+  output->linear_scale = 0.4;
+  output->rotational_scale = 0.8;
+  output->joint_scale = 0.5;
+  output->command_out_topic = "/fake_joint_trajectory_controller/joint_trajectory";
+  output->publish_period = 0.008;
+  output->command_out_type = "trajectory_msgs/JointTrajectory";
+  output->publish_joint_positions = true;
+  output->publish_joint_velocities = false;
+  output->publish_joint_accelerations = false;
+  output->joint_topic = "joint_states";
+  output->low_pass_filter_coeff = 2;
+  output->move_group_name = "panda_arm";
+  output->planning_frame = "panda_link0";
+  output->incoming_command_timeout = 0.1;
+  output->num_outgoing_halt_msgs_to_publish = 4;
+  output->lower_singularity_threshold = 17;
+  output->hard_stop_singularity_threshold = 30;
+  output->joint_limit_margin = 0.1;
+  output->check_collisions = true;
+  output->collision_check_rate = 50;
+  output->collision_check_type = "threshold_distance";
+  output->self_collision_proximity_threshold = 0.01;
+  output->scene_collision_proximity_threshold = 0.02;
+  output->collision_distance_safety_factor = 1000;
+  output->min_allowable_collision_distance = 0.01;
+
+  return output;
+}
+
+/** \brief Checks 2 ServoParameter objects for equality
+   */
+bool operator==(moveit_servo::ServoParameters& lhs, moveit_servo::ServoParameters& rhs)
```

Comment:
just wondering, any reason why you wanted to have this operator?

Reply:
Sure, it is purely for testing. While testing I have 2 ways to get parameters:

1. Through the parameters, as in normal launching (used in the integration tests)
2. Via the C++ call in the `test/test_parameter_struct.hpp` file - used in unit tests so we don't have to launch ROS stuff

During the parameters test, I do both and make sure they are equivalent. This is nice because the tests can only pass if all the parameters are the same and catches a case where somebody might change the yaml file and forget to update the parameter struct in that testing file. It also helped make sure the parameters were actually being loaded correctly